hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt

```diff
@@ -88,4 +89,47 @@ class PrisonerSearchByPrisonerNumbersResourceTest : AbstractSearchIntegrationTes
       .exchange()
       .expectStatus().isForbidden
   }
+
+  @Test
+  fun `returns only requested response fields`() {
+    webTestClient.post()
+      .uri {
+        it.path("/prisoner-search/prisoner-numbers")
+          .queryParam("responseFields", "prisonerNumber")
```

Comment:
would also be good to test for an empty list being sent as `responseFields` - as opposed to query param not being present

Reply:
> would also be good to test for an empty list being sent as `responseFields` - as opposed to query param not being present

This one is actually tested in `PrisonerResponseFieldsTest` / `should return all fields if empty list of fields requested`.